pkg/controller/statefulset/stateful_update_utils.go

```diff
@@ -0,0 +1,47 @@
+package statefulset
+
+import (
+	appsv1alpha1 "github.com/openkruise/kruise/pkg/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/util/priorityupdate"
+	v1 "k8s.io/api/core/v1"
+)
+
+func sortPodsToUpdate(rollingUpdateStrategy *appsv1alpha1.RollingUpdateStatefulSetStrategy, updateRevision string, replicas []*v1.Pod) []int {
+	var updateMin int
+	var priorityStrategy *appsv1alpha1.UpdatePriorityStrategy
+	if rollingUpdateStrategy != nil {
+		if rollingUpdateStrategy.Partition != nil {
+			updateMin = int(*rollingUpdateStrategy.Partition)
+		}
+		if rollingUpdateStrategy.UnorderedUpdate != nil {
+			priorityStrategy = rollingUpdateStrategy.UnorderedUpdate.PriorityStrategy
+		}
+	}
+	maxUpdate := len(replicas) - updateMin
+	if maxUpdate <= 0 {
+		return []int{}
+	}
+
+	var updatedIdxs []int
+	var waitUpdateIdxs []int
```

Comment:
is it better to initialize Idxs slice with capacity equals to length of replicas

Reply:
This is unnecessary, i think.